core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -103,6 +122,7 @@ class KafkaApis(val requestChannel: RequestChannel,
     // We can't have the ensureTopicExists check here since the controller sends it as an advisory to all brokers so they
     // stop serving data to clients for the topic being deleted
     val leaderAndIsrRequest = request.requestObj.asInstanceOf[LeaderAndIsrRequest]
+    KafkaApis.validateRequestVersion(request, Option(leaderAndIsrRequest.versionId))
```

Comment:
Minor, but It's better to use `Some` instead of `Option` for cases where we are not passing a potentially null value. There are other similar examples in this class.